pkg/legacy/tarball/write_test.go

```diff
@@ -456,3 +460,147 @@ func TestUncompressedSize(t *testing.T) {
 		t.Errorf("expected UncompressedSize to be called, but it wasn't")
 	}
 }
+
+func TestWriteSharedLayers(t *testing.T) {
```

Comment:
Can you add a comment describing the scenario this tests?

Something like: "TestWriteSharedLayers tests that writing a tarball of multiple images that share some layers only writes those shared layers once." (feel free to reword)

I'm not sure I'd understand this test at a glance outside the context of this PR and a comment would help a lot.